crates/uv-python/src/discovery.rs

```diff
@@ -908,11 +908,26 @@ pub(crate) fn find_python_installation(
             continue;
         }
 
+        // If it's an alternative implementation, and alternative implementations aren't allowed
+        // skip it
+        if !matches!(
+            installation.implementation(),
+            LenientImplementationName::Known(ImplementationName::CPython),
+        ) && !request.allows_alternative_implementations()
+            && !installation.source.allows_alternative_implementations()
+            // If found on the search path with the default executable name we should not skip it
+            && !(installation.uses_default_executable_name()
+                && installation.source == PythonSource::SearchPath)
```

Comment:
This is a bit annoying, but necessary to prevent a regression when you have an alternative implementation on your PATH as `python`. We might want to track executable names in `PythonSource::SearchPath` directly?